pandas/types/inference.py

```diff
@@ -35,8 +35,28 @@ def _iterable_not_string(x):
 
 
 def is_iterator(obj):
-    # python 3 generators have __next__ instead of next
-    return hasattr(obj, 'next') or hasattr(obj, '__next__')
+    if not hasattr(obj, '__iter__'):
+        return False
+
+    if PY2:
+        return hasattr(obj, 'next')
+    else:
+        # Python 3 generators have
+        # __next__ instead of next
+        return hasattr(obj, '__next__')
+
+
+def is_file_like(obj):
+    file_attrs = ('read', 'write', 'seek', 'tell')
```

Comment:
can you add a doc-string describing this. (and of course any others you want to document; prob should open an issue for that).